src/app/component/ProductCompare/ProductCompare.component.js

```diff
@@ -0,0 +1,152 @@
+              mix={ { block: 'ClearButton' } }
+            >
+                <button onClick={ clearCompareList }>
+                    { __('Clear Compare') }
+                </button>
+            </div>
+        );
+    }
+
+    renderProductCards() {
+        const { products } = this.props;
+
+        return products.map((product) => (
+            <div block="ProductCompare" elem="Item" key={ product.id }>
+                <ProductCompareItem product={ product } />
+            </div>
+        ));
+    }
+
+    renderPriceLabel() {
+        return (
+            <div
+              block="ProductCompare"
+              elem="FirstColumn"
+              mix={ { block: 'PriceLabel' } }
+            >
+                { __('Price') }
+            </div>
+        );
+    }
+
+    renderProductPrices() {
+        const { products } = this.props;
+
+        return products.map(({ id, price_range }) => <ProductPrice price={ price_range } key={ id } />);
+    }
+
+    renderAttributes() {
+        const { getAttributes } = this.props;
+        const attributes = getAttributes();
+
+        return attributes.map(({ attribute_id, attribute_label, attribute_values }) => (
+            <ProductCompareAttributeRow
+              title={ attribute_label }
+              values={ attribute_values }
+              key={ attribute_id }
+            />
+        ));
+    }
+
+    renderProducts() {
+        const { products, device } = this.props;
+        const wideItems = !device.isMobile && products.length < MAX_DISPLAY_ITEMS_BEFORE_OVERFLOW;
+
+        return (
+            <div block="ProductCompare" mods={ { wideItems } }>
+                <div
+                  block="ProductCompare"
+                  elem="Row"
+                  mix={ { block: 'ProductCardRow' } }
+                >
+                    { this.renderClearButton() }
+                    { this.renderProductCards() }
+                </div>
+                <div
+                  block="ProductCompare"
+                  elem="Row"
+                  mix={ { block: 'ProductPriceRow' } }
+                >
+                    { this.renderPriceLabel() }
+                    { this.renderProductPrices() }
+                </div>
+                <div
+                  block="ProductCompare"
+                  elem="AttributeTable"
+                >
+                    { this.renderAttributes() }
+                </div>
+            </div>
+        );
+    }
+
+    renderEmpty() {
+        return <div block="ProductCompare" elem="Empty">{ __('You have nothing to compare') }</div>;
+    }
+
+    render() {
+        const {
+            isLoading,
+            products
+        } = this.props;
+        const hasProducts = products && products.length;
+
+        if (isLoading && !hasProducts) {
+            return null;
+        }
+
+        if (!hasProducts) {
+            return this.renderEmpty();
+        }
+
+        return this.renderProducts();
```

Comment:
You indended it to show product when loading, if not empty?

Reply:
Fixed